controllers/clusterclassutil/indexer.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package clusterclassutil implements ClusterClass utils
+package clusterclassutil
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AddClusterClassNameIndex adds the cluster topology class index to the
+// manager cache.
+func AddClusterClassNameIndex(ctx context.Context, mgr ctrl.Manager) error {
+	if err := mgr.GetCache().IndexField(ctx, &clusterv1.Cluster{},
+		clusterv1.ClusterClassNameIndex,
+		indexClusterByClassName,
+	); err != nil {
+		return errors.Wrap(err, "error setting index fields")
```

Comment:
```suggestion
		return errors.Wrap(err, "error setting index field")
```
nit

Reply:
Fun fact: looks like the IndexField func was previously named IndexFields in controller-runtime (or at least the go doc in controller-runtime uses the wrong func name)